controller/src/main/java/io/pravega/controller/fault/SegmentMonitorLeader.java

```diff
@@ -171,14 +168,14 @@ public void takeLeadership(CuratorFramework client) throws Exception {
     }
 
     /**
-     * Blocks until the rebalance timer is zero so that we honor the minimum rebalance interval.
+     * Blocks until the rebalance interval. This wait serves multiple purposes:
+     * -- Ensure rebalance does not happen in quick succession since its a costly cluster level operation.
+     * -- Clubs multiple host events into one to reduce rebalance operations. For example:
+     *      Fresh cluster start, cluster/multi-host/host restarts, etc.
      */
     private void waitForRebalance() throws InterruptedException {
-        if (timeoutTimer != null && timeoutTimer.getRemaining().getSeconds() > 0) {
-            log.info("Waiting for {} seconds before attempting to rebalance",
-                    timeoutTimer.getRemaining().getSeconds());
-            Thread.sleep(timeoutTimer.getRemaining().getSeconds() * 1000);
-        }
+        log.info("Waiting for {} seconds before attempting to rebalance", minRebalanceInterval.getSeconds());
```

Comment:
It looks like this call sleeps for `minRebalanceInterval` no matter what. Is there any chance this can be invoked by multiple threads? I don't think it makes sense to have multiple threads sleeping here, so I'm trying to understand the thread situation here and whether we need to make it thread safe. Otherwise, we need to make sure it is marked as not thread safe.